### PR TITLE
removing transaction committing node

### PR DESCRIPTION
**Pull request by jycor**
doltgres fix: https://github.com/dolthub/doltgresql/pull/872

### CODE REVIEW

File: engine.go [hunk 1]
```diff
@@ -722,31 +724,6 @@ func (e *Engine) CloseSession(connID uint32) {
 	e.PreparedDataCache.DeleteSessionData(connID)
 }
 
-// Count number of BindVars in given tree
-func countBindVars(node sql.Node) int {
```
Comment (max-hoffman) [inline engine.go:726]: this is unrelated, but unused code?

File: engine.go [hunk 2]
```diff
@@ -446,6 +446,8 @@ func (e *Engine) QueryWithBindings(ctx *sql.Context, query string, parsed sqlpar
 
 		return nil, nil, nil, err
 	}
+
+	iter = rowexec.AddTransactionCommittingIter(iter, qFlags)
```
Comment (max-hoffman) [inline engine.go:450]: if we keep doing this, it might make sense to group into a helper function with a doc comment

**Review by max-hoffman — APPROVED** [review]
lgtm

File: engine.go (hunk 1 above)
Comment (jycor) [reply engine.go:726]: yeah unused code

**Pull request merged**